external-crates/move/crates/move-compiler/tests/move_2024/expansion/or_pat_mutability_mismatch.exp

```diff
@@ -0,0 +1,28 @@
+error[E03015]: invalid pattern
+  ┌─ tests/move_2024/expansion/or_pat_mutability_mismatch.move:9:18
+  │
+9 │             X::A(mut x) | X::B(x, _) => {
+  │                  ^^^      ---------- right or-pattern binds it immutably
+  │                  │         
+  │                  left or-pattern binds variable x mutably
+  │
+  = Both sides of an or-pattern must bind the same variables with the same mutability.
+
+error[E03015]: invalid pattern
+   ┌─ tests/move_2024/expansion/or_pat_mutability_mismatch.move:18:28
+   │
+18 │             X::A(x) | X::B(mut x, _) => {
+   │                  -         ^^^ right or-pattern binds variable x mutably
+   │                  │          
+   │                  left or-pattern binds it immutably
+   │
+   = Both sides of an or-pattern must bind the same variables with the same mutability.
+
+error[E04024]: invalid usage of immutable variable
+   ┌─ tests/move_2024/expansion/or_pat_mutability_mismatch.move:19:17
+   │
+18 │             X::A(x) | X::B(mut x, _) => {
```

Comment:
It's weird that we get this error here. I would expect to mark the variable as mutable so typing doesn't generate this error.